CRM/Campaign/Form/Petition/Signature.php

```diff
@@ -371,17 +371,7 @@ public function postProcess() {
       $ids[0] = $this->_contactId;
     }
     else {
-      // dupeCheck - check if contact record already exists
-      // code modified from api/v2/Contact.php-function civicrm_contact_check_params()
-      $params['contact_type'] = $this->_ctype;
-      //TODO - current dedupe finds soft deleted contacts - adding param is_deleted not working
-      // ignore soft deleted contacts
-      //$params['is_deleted'] = 0;
-      $dedupeParams = CRM_Dedupe_Finder::formatParams($params, $params['contact_type']);
-      $dedupeParams['check_permission'] = '';
-
-      //dupesByParams($params, $ctype, $level = 'Unsupervised', $except = array())
-      $ids = CRM_Dedupe_Finder::dupesByParams($dedupeParams, $params['contact_type']);
+      $ids = CRM_Contact_BAO_Contact::getDuplicateContacts($params, $this->_ctype, 'Unsupervised', array(), FALSE);
```

Comment:
@seamuslee001 just to talk through this change - there are a few of these (not all submitted as yet) to make & obviously I want to get them right but it's really hard to figure out how to test all places!

formatParams is now in getDuplicateContacts so the $params formerly passed to that function needs to go there. Otherwise the parameters for getDuplicateContacts match CRM_Dedupe_Finder::dupesByParams except the last parameter if check_permission 

getDuplicateContacts($input, $contactType, $rule = 'Unsupervised', $excludedContactIDs = array(), $checkPermissions = TRUE, $ruleGroupID = NULL)

I guess in the above I need to double check removing $params['contact_type'] = $this->_ctype; will not cause a problem